File: MySQLtest.go
package main

import (
	"database/sql"
	"fmt"
	"log"

	_ "github.com/go-sql-driver/mysql"
)

//Newmenu is struct
type Newmenu struct {
	id         int
	name       string
	price      int
	starttime  string
	lastupdate string
}

//NewMysql is struct
type NewMysql struct {
	tablename string
	address   string
	protocol  string
	dbname    string
	user      string
	pass      string
}

func main() {

	val := Newmenu{name: "タバスコいりカクテル", price: 800, starttime: "2015-3-20"}
	opt := NewMysql{tablename: "menu", dbname: "nyumon", user: "nyuser", pass: "nypass"}

	//今回はlocalhostなのでuser名:password@/db名でOK
	db, err := sql.Open("mysql", opt.user+":"+opt.pass+"@/"+opt.dbname) //opt.user+":"+opt.pass+"@"+opt.protocol+"("+opt.address+")/"+opt.db
	log.Println("Connected to mysql.")

	if err != nil {
		log.Fatal(err)
	}
	//上のerrより下のif errの方がいいらしい
	err = db.Ping()
	if err != nil {
		panic(err)
	}
	defer db.Close()

	//INSERT
	S := fmt.Sprintf("INSERT INTO %s (name,price,start_date) VALUES(?, ?, ?);", opt.tablename)
	ins, err := db.Prepare(S)
	if err != nil {
		log.Fatal(err)
	}
	defer ins.Close()
	ins.Exec(val.name, val.price, val.starttime)

	//SELECT
	rows, err := db.Query("SELECT * FROM " + opt.tablename + ";")
	defer rows.Close()
	if err != nil {
		log.Fatal(err)
	}

	var allresult []Newmenu
	for rows.Next() {
		returnmenu := Newmenu{}
		err := rows.Scan(&returnmenu.id, &returnmenu.name, &returnmenu.price, &returnmenu.starttime, &returnmenu.lastupdate)
		if err != nil {
			log.Fatal(err)
		}
		allresult = append(allresult, returnmenu)
	}

	for _, u := range allresult {
		fmt.Printf("ID:%d name:%s price:%d start_date:%s lastupdate:%s\n", u.id, u.name, u.price, u.starttime, u.lastupdate)
	}

}

File: MySQLtest.py
import mysql.connector
from urllib.parse import urlparse

def mysql_inserts(cur_name,tablename,insert_data_list):
    sql_str = "INSERT INTO {} (name,price,start_date) VALUES (%s, %s, %s)".format(tablename)
    cur_name.executemany(sql_str,insert_data_list)

def mysql_insert(cur_name,tablename,data):
    if type(data) is dict:
        sql_str = "INSERT INTO {} (name,price,start_date) VALUES (%(name)s, %(price)s, %(start_date)s)".format(tablename)
        cur_name.execute(sql_str, data)
    else:
        sql_str = "INSERT INTO {} (name,price,start_date) VALUES (%s, %s, %s)".format(tablename)
        cur_name.execute(sql_str, data)

def mysql_select(cur_name,tablename):
    sql_str = "SELECT * FROM {}".format(tablename)
    cur_name.execute(sql_str)
    rows = cur_name.fetchall()
    for row in rows:
        print(row)


#localhostのMySQLのDB名:nyumonにユーザー名:ex_user password:ex_passで接続し,テーブル名:menuに対して操作する
def main():
    mysql_url = urlparse("mysql://ex_user:ex_pass@localhost:3306/nyumon")#デフォルトが3306

    conn = mysql.connector.connect(
        host = mysql_url.hostname or "localhost",
        port = mysql_url.port or 3306,
        user = mysql_url.username or "root",
        password = mysql_url.password or " ",
        database = mysql_url.path[1:]
    )
    print(conn.is_connected())
    conn.ping(reconnect=True)

    cur = conn.cursor()#カーソル

    tablename = "menu"
    data_tup = ("醤油カクテル",1100,"2009-1-11")
    data_list = ["抹茶カクテル", 850, "2007-7-8"]
    data_dic = {"name":"生姜カクテル", "price":725, "start_date":"2012-5-29"}
    datas = [
        ("オレンジ", 500, "2000-1-1"),
        ("リンゴ", 400, "2000-1-1"),
        ("桃", 300, "2000-1-1")
    ]


    mysql_insert(cur,tablename,data_list)
    mysql_insert(cur,tablename,data_tup)
    mysql_insert(cur,tablename,data_dic)
    mysql_inserts(cur,tablename,datas)
    mysql_select(cur,tablename)

    conn.commit()#しないと変更が反映されなかった。
    cur.close()
    conn.close()

if __name__ == "__main__":
    main()
